### PR TITLE
Push down RPC reqWallet flag

### PR DESCRIPTION
**Pull request by jonasschnelli**

### CODE REVIEW

**Comment by sipa** [comment]
Concept ACK.

**Comment by jgarzik** [comment]
ut ACK

Would prefer code movement in a first commit, separated out from the rest.

**Comment by jonasschnelli** [comment]
separated out the code moving

**Comment by jgarzik** [comment]
Looks good.  This probably wants a release note, as it's a slight RPC API-visible change.

**Comment by sipa** [comment]
How is it visible?

**Comment by jgarzik** [comment]
An exception is no longer thrown in all cases. Value::null return value comes through.

**Comment by jonasschnelli** [comment]
This should not be the case. Value::null should only come through for the help string generation (in -disablewallet mode the cmd should not be listed under help). I tried to avoid any, even slightly, API changes.

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -2118,3 +2232,115 @@ Value resendwallettransactions(const Array& params, bool fHelp)
...
+    }
+    
+    Array results;
+    vector<COutput> vecOutputs;
+    assert(pwalletMain != NULL);
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+    pwalletMain->AvailableCoins(vecOutputs, false);
+    BOOST_FOREACH(const COutput& out, vecOutputs) {
+        if (out.nDepth < nMinDepth || out.nDepth > nMaxDepth)
+        continue;
```
Comment (laanwj) [inline src/wallet/rpcwallet.cpp]: Something weird happened to the indentation of this function, for example these continues are not indented.

Comment (jonasschnelli) [inline src/wallet/rpcwallet.cpp]: IDE issue! Will fix this.

Comment (jonasschnelli) [inline src/wallet/rpcwallet.cpp]: Fixed.

File: src/rpcserver.cpp [hunk 2]
```diff
@@ -193,11 +193,6 @@ string CRPCTable::help(string strCommand) const
             continue;
         if ((strCommand != "" || pcmd->category == "hidden") && strMethod != strCommand)
             continue;
-#ifdef ENABLE_WALLET
```
Comment (laanwj) [inline src/rpcserver.cpp]: This means that the wallet commands will be listed in help, even if the wallet is disabled. Is that on purpose?

**Comment by laanwj** [comment]
> Value::null should only come through for the help string generation (in -disablewallet mode the cmd should not be listed under help). I

The help flag parameter is set to true when a full help message is requested, in case of `help <command>`. In this case throwing an exception is fine.

File: src/rpcserver.cpp (hunk 2 above)
Comment (jonasschnelli) [inline src/rpcserver.cpp]: No.
the `!pwalletMain` check (wallet disabled at runtime yes/no) will be done within the new function `EnsureWalletIsAvailable(bool)` :
https://github.com/bitcoin/bitcoin/pull/5992/files#diff-df7d84ff2f53fcb2a0dc15a3a51e55ceR40

Comment (laanwj) [inline src/rpcserver.cpp]: Ok, fair enough. Forget my other remark then.

**Comment by laanwj** [comment]
Tested ACK

**Pull request merged**